library/std/src/fs.rs

```diff
@@ -615,6 +615,181 @@ impl File {
         self.inner.datasync()
     }
 
+    /// Acquire an exclusive advisory lock on the file. Blocks until the lock can be acquired.
+    ///
+    /// This acquires an exclusive advisory lock; no other file handle to this file may acquire
+    /// another lock.
+    ///
+    /// If this file handle already holds an advisory lock the exact behavior is
```

Comment:
If you dup a file handle is the result considered a separate file handle or are both considered a single file handle for the purposes of this api?

Reply:
They're the same. I will clarify this